common/src/bitset.rs

```diff
@@ -302,6 +308,67 @@ impl BitSet {
     pub fn tinyset(&self, bucket: u32) -> TinySet {
         self.tinysets[bucket as usize]
     }
+
+    /// next_set_bit returns the lowest set bit >= `start`, or -1 if none.
+    /// Returns the lowest set bit >= `start`, or `u32::MAX` if none are set.
+    pub fn next_set_bit(&self, start: u32) -> u32 {
+        if start >= self.max_value {
+            return u32::MAX;
+        }
+        let mut bucket = (start / 64) as usize;
+        let mut bit_in_bucket = start % 64;
+        while bucket < self.tinysets.len() {
+            let ts = self.tinysets[bucket].0;
+            // zero out bits lower than `bit_in_bucket`
+            let mask = ts & (!((1 << bit_in_bucket) - 1));
```

Comment:
the mask is the right part. 

ts & mask is the masked version of ts.

Reply:
The code would benefit from extracting that logic and put it in an line method of TinySet.